packages/notionable/src/global/sagas/on-data-update.ts

```diff
@@ -107,3 +108,26 @@ function onActiveAccounts$(global$: Observable<GlobalState>) {
     })
   );
 }
+
+function onHistoricalTrading$(global$: Observable<GlobalState>) {
+  return global$.pipe(
+    distinctUntilChanged(
+      (p, c) =>
+        isAppReady(p.networkState) === isAppReady(c.networkState) &&
+        c.baseCurrency === p.baseCurrency
+    ),
+    filter((g) => isAppReady(g.networkState)),
+    switchMap(() => {
+      return timer(500, 60_000).pipe(
+        map(() => ({
+          historicalTrading: SupportedNetworks.reduce((acc, n) => {
+            if (Registry.getAnalyticsRegistry().isNetworkRegistered(n)) {
+              acc[n] = Registry.getAnalyticsRegistry().getHistoricalTrading(n);
```

Comment:
I would process the liquidity pool table data for all currency ids at this level and then you can just read them out of global state.